pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/AsciidoctorService.java

```diff
@@ -324,6 +322,17 @@ private String buildDocument(@Nonnull Document base, @Nonnull Locale locale, @No
                 content.append(xrefProcessor.preprocess(rawContent));
 
                 html = asciidoctor.convert(content.toString(), ob.get());
+                Violations violations = new XrefValidator(html).validate();
+                Validations validationResult = documentVersion.validations().getOrCreate();
```

Comment:
This can be a part of of utility method of a Helper class, for example `ValidationDataOps` such that the Violations can be passed and  the validation nodes will be returned.

Reply:
@aprajshekhar Apart from Violations, we need to pass the context (in our case documentVersion) to create and return Validation node.